models/actions/run.go

```diff
@@ -308,6 +308,17 @@ func InsertRun(ctx context.Context, run *ActionRun, jobs []*jobparser.SingleWork
 	return commiter.Commit()
 }
 
+func GetLatestRunForBranchAndWorkflow(ctx context.Context, repoID int64, branch, workflowFile string) (*ActionRun, error) {
+	var run ActionRun
+	has, err := db.GetEngine(ctx).Where("repo_id=?", repoID).And("ref=?", branch).And("workflow_id=?", workflowFile).Desc("id").Get(&run)
+	if err != nil {
+		return nil, err
+	} else if !has {
+		return nil, util.NewNotExistErrorf("run with repo_id %d, ref %s, workflow_id %s", repoID, branch, workflowFile)
+	}
+	return &run, nil
+}
+
```

Comment:
But the `schedule` run doesn't have a branch ref?

Reply:
The function's purpose is - as the name implies - get a Run for a given branch and workflow. It is not concerned with runs that don't have a branch.

Reply:
But in your route, `workflows/{workflow_name}/status.svg` uses this underlying function `GetLatestRunForBranchAndWorkflow` with the default branch. If a workflow file that runs can only be triggered by schedule, then no badge will be returned. But IMO, `workflows/{workflow_name}/status.svg` should return the last run status.

Reply:
Mmm. That certainly wasn't the use-case for the route I had in mind, but I see your point. Will see what I can do. Perhaps it'd check if a workflow exists for the default branch, and if not, select the latest scheduled one. I suppose that'd satisfy both needs.

Alternatively - and I think I'd like that approach better - I can change the routes:

- `workflows/{workflow_name}/status.svg` would check the main branch.
- `workflows/{workflow_name}/branch/{branch_name}/status.svg` would check a given branch.
- `workflows/{workflow_name}/scheduled/status.svg` would check the latest scheduled run.

What do you think?

Reply:
A scheduled run does have a branch ref if I understand it correctly - it is just sometimes empty due to a bug which should be fixed with #28640

Reply:
Now scheduled run has a default branch ref, I think the router can be designed to be compatible with GitHub.
https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge
workflows/{workflow_name}/badge.svg with `event` and `branch` para.

Reply:
Can do, I'll update the PR later today.

Reply:
Hrm. I *can* make it compatible with GitHub, even the same URI, just different host. But I kinda like how every badge lives under `/badges/`. I can make both work, though. I suppose I'll do that, and let others decide which one to keep (or to keep both).